### DISCUSSION TITLE
Requesting key recovery by certificate owners?

### DISCUSSION THREAD

**Discussion by dledob** [Keyfactor/ejbca-ce#605] [Q&A]
It is possible that the owners of the certificates can request the key recovery through the RA GUI? or this possibility is only for administrators with the correct privileges.

**Comment by primetomas** [comment]
I think this is possible by using Approvals.
https://docs.keyfactor.com/ejbca/latest/approvals

**Reply by dledob** [reply]
when one accesses the RA GUI without a client certificate, there are no options available due to permissions
So what does a user do if they lost their secret key? How to authenticate to request key recovery?
The same thing happens if I want to request that my certificate be revoked.
Are there those functionalities in the RA GUI? If they don't exist, it would be interesting to add them.

**Reply by primetomas** [reply]
You can do a lot with approvals, giving some rights to unauthenticated users to perform actions that then require approvals. See the example work flows in the documentation for approvals. I’m not sure if you can do exactly what you ask for though.
